### DISCUSSION TITLE
Prisma Adapter + Mysql + Unable to SignIn

### DISCUSSION THREAD

**Discussion by sudoak** [nextauthjs/next-auth#3852] [Help]
### Title

Prisma Adapter + Mysql + Unable to SignIn

### How to reproduce ☕️

I'm running the boilerplate NextAuth setup with the Prisma adapter connecting to a Mysql db in localhost. On Google signin the following error occurs which I tried resolving by changing the account model with no success.

### Your question/bug report 📓

PS C:\Users\IIntel\Documents\code\event-timeline> npm run dev

> dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from C:\Users\IIntel\Documents\code\event-timeline\.env.local
info  - Loaded env from C:\Users\IIntel\Documents\code\event-timeline\.env
event - compiled client and server successfully in 581 ms (656 modules)
wait  - compiling / (client and server)...
event - compiled client and server successfully in 268 ms (1191 modules)
Warning: <title> should not be used in _document.js's <Head>. https://nextjs.org/docs/messages/no-document-title
wait  - compiling /api/auth/[...nextauth]...
event - compiled client and server successfully in 109 ms (1206 modules)
prisma:query SELECT `test_db`.`Account`.`id`, `test_db`.`Account`.`userId` FROM `test_db`.`Account` WHERE (`test_db`.`Account`.`provider` = ? AND `test_db`.`Account`.`providerAccountId` = ?) LIMIT ? OFFSET ?
prisma:query SELECT `test_db`.`Account`.`id`, `test_db`.`Account`.`userId` FROM `test_db`.`Account` WHERE (`test_db`.`Account`.`provider` = ? AND `test_db`.`Account`.`providerAccountId` = ?) LIMIT ? OFFSET ?
prisma:query SELECT `test_db`.`User`.`id`, `test_db`.`User`.`name`, `test_db`.`User`.`email`, `test_db`.`User`.`emailVerified`, `test_db`.`User`.`image` FROM `test_db`.`User` WHERE `test_db`.`User`.`email` = ? LIMIT ? OFFSET ?
prisma:query BEGIN
prisma:query INSERT INTO `test_db`.`User` (`id`,`name`,`email`,`emailVerified`,`image`) VALUES (?,?,?,?,?)
prisma:query SELECT `test_db`.`User`.`id`, `test_db`.`User`.`name`, `test_db`.`User`.`email`, `test_db`.`User`.`emailVerified`, `test_db`.`User`.`image` FROM `test_db`.`User` WHERE `test_db`.`User`.`id` = ? LIMIT ? OFFSET ?
prisma:query COMMIT
prisma:query BEGIN
prisma:query INSERT INTO `test_db`.`Account` (`id`,`userId`,`type`,`provider`,`providerAccountId`,`refresh_token`,`accessToken`,`expires_at`,`token_type`,`scope`,`id_token`) VALUES (?,?,?,?,?,?,?,?,?,?,?)
prisma:query SELECT `test_db`.`Account`.`id`, `test_db`.`Account`.`userId`, `test_db`.`Account`.`type`, `test_db`.`Account`.`provider`, `test_db`.`Account`.`providerAccountId`, `test_db`.`Account`.`refresh_token`, `test_db`.`Account`.`accessToken`, `test_db`.`Account`.`expires_at`, `test_db`.`Account`.`token_type`, `test_db`.`Account`.`scope`, `test_db`.`Account`.`id_token`, `test_db`.`Account`.`session_state`, `test_db`.`Account`.`oauth_token_secret`, `test_db`.`Account`.`oauth_token` FROM `test_db`.`Account` WHERE `test_db`.`Account`.`id` = ? LIMIT ? OFFSET ?
prisma:query COMMIT
prisma:query BEGIN
prisma:query INSERT INTO `test_db`.`Session` (`id`,`sessionToken`,`userId`,`expires`) VALUES (?,?,?,?)
prisma:query SELECT `test_db`.`Session`.`id`, `test_db`.`Session`.`sessionToken`, `test_db`.`Session`.`userId`, `test_db`.`Session`.`expires` FROM `test_db`.`Session` WHERE `test_db`.`Session`.`id` = ? LIMIT ? OFFSET ?
prisma:query COMMIT
Warning: <title> should not be used in _document.js's <Head>. https://nextjs.org/docs/messages/no-document-title
prisma:query SELECT `test_db`.`Session`.`id`, `test_db`.`Session`.`sessionToken`, `test_db`.`Session`.`userId`, `test_db`.`Session`.`expires` FROM `test_db`.`Session` WHERE `test_db`.`Session`.`sessionToken` = ? LIMIT ? OFFSET ?
prisma:query SELECT `test_db`.`User`.`id`, `test_db`.`User`.`name`, `test_db`.`User`.`email`, `test_db`.`User`.`emailVerified`, `test_db`.`User`.`image` FROM `test_db`.`User` WHERE `test_db`.`User`.`id` IN (?)
[next-auth][error][SESSION_ERROR]
https://next-auth.js.org/errors#session_error Cannot read properties of undefined (reading 'accessToken') {
  message: "Cannot read properties of undefined (reading 'accessToken')",
  stack: "TypeError: Cannot read properties of undefined (reading 'accessToken')\n" +
    '    at Object.session (webpack-internal:///./pages/api/auth/[...nextauth].tsx:46:41)\n' +
    '    at Object.session (C:\\Users\\IIntel\\Documents\\code\\event-timeline\\node_modules\\next-auth\\core\\routes\\session.js:111:48)\n' +
    '    at async NextAuthHandler (C:\\Users\\IIntel\\Documents\\code\\event-timeline\\node_modules\\next-auth\\core\\index.js:96:27)\n' +
    '    at async NextAuthNextHandler (C:\\Users\\IIntel\\Documents\\code\\event-timeline\\node_modules\\next-auth\\next\\index.js:20:19)\n' +
    '    at async C:\\Users\\IIntel\\Documents\\code\\event-timeline\\node_modules\\next-auth\\next\\index.js:56:32\n' +
    '    at async Object.apiResolver (C:\\Users\\IIntel\\Documents\\code\\event-timeline\\node_modules\\next\\dist\\server\\api-utils.js:102:9)\n' +
    '    at async DevServer.handleApiRequest (C:\\Users\\IIntel\\Documents\\code\\event-timeline\\node_modules\\next\\dist\\server\\base-server.js:1076:9)\n' +
    '    at async Object.fn (C:\\Users\\IIntel\\Documents\\code\\event-timeline\\node_modules\\next\\dist\\server\\base-server.js:963:37)\n' +
    '    at async Router.execute (C:\\Users\\IIntel\\Documents\\code\\event-timeline\\node_modules\\next\\dist\\server\\router.js:222:32)\n' +
    '    at async DevServer.run (C:\\Users\\IIntel\\Documents\\code\\event-timeline\\node_modules\\next\\dist\\server\\base-server.js:1103:29)',
  name: 'TypeError'
}

### Contributing  🙌🏽

No, I'm afraid I cannot help regarding this

**Comment by ndom91** [comment]
`nextauthjs/next-auth` has been migrated to a monorepository. The adapters code can now be found there under `packages/adapter-*`. Thanks for your interest in the project!

**Comment by ndom91** [comment]
Hi @sudoak can you please provide us with some more detailed information? Like your `[...nextauth].js` configuration? Thanks!

**Comment by sudoak** [comment]
Hi @ndom91, I followed your above mentioned steps, still getting error.

![Screenshot 2022-02-05 222923](https://user-images.githubusercontent.com/25544768/152651060-54015936-5824-4523-afa0-57ba3c12edc6.png)

![Screenshot 2022-02-05 223006](https://user-images.githubusercontent.com/25544768/152651064-158dd1ea-d7d0-456d-842e-ad48762ffffa.png)

**Reply by ndom91** [reply]
It doesn't look like you've changed the field type of the id_token field

**Comment by ndom91** [comment]
> Hi @ndom91, I followed your above mentioned steps, still getting error.
> 
> ![Screenshot 2022-02-05 222923](https://user-images.githubusercontent.com/25544768/152651060-54015936-5824-4523-afa0-57ba3c12edc6.png)
> 
> ![Screenshot 2022-02-05 223006](https://user-images.githubusercontent.com/25544768/152651064-158dd1ea-d7d0-456d-842e-ad48762ffffa.png)
> 

Yeah now you have a different error, your trying to put too much info in the `id_token` and it becomes too long for your database to save with the configured field type. 

It seems your using MySQL, right? Prisma seems to map this `String` type from the schema to `varchar(191)` in MySQL under the hood (see: https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#string). 

You may have to either manually change this `id_token` field type to something dynamic or longer, or try to save less in it. 

**Edit**: for example, you could change the prisma schema line for `id_token` to the following: 

``` 
id_token String? @db.LongText 
``` 

In order to map it to a MySQL `LONGTEXT` type, which should definitely be able to hold longer strings. 

Don't forget to reapply the schema after making this change.

**Reply by sudoak** [reply]
Hi @ndom91, earlier issue cropped up

![newError](https://user-images.githubusercontent.com/25544768/152686488-f8f50280-cc0c-4e1e-9de0-fa14afd01540.png)

**Reply by ndom91** [reply]
@sudoak can you show us your currently applied schema as well as your `[...nextauth].js` configuration? Otherwise its tough to say exactly whats going wrong.

**Comment by sudoak** [comment]
Hi @ndom91, below are the required files

[...nextauth].ts -> https://gist.github.com/sudoak/76c6bb57cdc4de3f3accfd93d0e80f5d
schema.prisma -> https://gist.github.com/sudoak/f8d06aa4f82132c08ff3ba306a62354e

**Reply by ndom91** [reply]
Thanks, will take a look later on :+1:

**Comment by Unieveth** [comment]
Hi, I am also having the same issue as @sudoak .  I followed the setup steps at https://next-auth.js.org/adapters/prisma and now I can login but I don't have a session to use.

```
[next-auth][error][SESSION_ERROR]
https://next-auth.js.org/errors#session_error Cannot read properties of undefined (reading 'accessToken') {
  message: "Cannot read properties of undefined (reading 'accessToken')",
  stack: "TypeError: Cannot read properties of undefined (reading 'accessToken')\n" +
    '    at Object.session (webpack-internal:///(api)/./pages/api/auth/[...nextauth].ts:41:41)\n' +
    '    at Object.session (/home/unieveth/uni-overlay/node_modules/next-auth/core/routes/session.js:111:48)\n' +
    '    at async NextAuthHandler (/home/unieveth/uni-overlay/node_modules/next-auth/core/index.js:158:27)\n' +
    '    at async NextAuthNextHandler (/home/unieveth/uni-overlay/node_modules/next-auth/next/index.js:23:19)\n' +
    '    at async /home/unieveth/uni-overlay/node_modules/next-auth/next/index.js:59:32\n' +
    '    at async Object.apiResolver (/home/unieveth/uni-overlay/node_modules/next/dist/server/api-utils/node.js:366:9)\n' +
    '    at async DevServer.runApi (/home/unieveth/uni-overlay/node_modules/next/dist/server/next-server.js:469:9)\n' +
    '    at async Object.fn (/home/unieveth/uni-overlay/node_modules/next/dist/server/next-server.js:719:37)\n' +
    '    at async Router.execute (/home/unieveth/uni-overlay/node_modules/next/dist/server/router.js:247:36)\n' +
    '    at async DevServer.run (/home/unieveth/uni-overlay/node_modules/next/dist/server/base-server.js:346:29)',
  name: 'TypeError'
}
```

If this was fixed I cant seem to find an fix on Github

**Reply by dEvAshirvad** [reply]
Did you get the solution broo...
i am stuck at the same issue

**Comment by hakrichTech** [comment]
Anyone with the solution about this bro ...
